### PR TITLE
[Improve][formats]Unified exception for seatunnel formats module

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3571
Unified exception for seatunnel formats module
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonDeserializationSchema.java [hunk 1]
```diff
@@ -70,7 +71,7 @@ public JsonDeserializationSchema(boolean failOnMissingField,
                                      boolean ignoreParseErrors,
                                      SeaTunnelRowType rowType) {
         if (ignoreParseErrors && failOnMissingField) {
-            throw new IllegalArgumentException(
+            throw new SeaTunnelRuntimeException(CommonErrorCode.ILLEGAL_ARGUMENT,
```
Comment (TyrantLucifer) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonDeserializationSchema.java]: UNSOPPORTED_OPERATION is better.

**Review by TyrantLucifer — COMMENTED** [review]
BTW, for every format modules create their own exception is better.

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonDeserializationSchema.java (hunk 1 above)
Comment (TaoZex) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonDeserializationSchema.java]: Thanks.Done.

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonSerializationSchema.java [hunk 2]
```diff
@@ -60,8 +62,8 @@ public byte[] serialize(SeaTunnelRow row) {
             runtimeConverter.convert(mapper, node, row);
             return mapper.writeValueAsBytes(node);
         } catch (Throwable e) {
-            throw new RuntimeException(
-                String.format("Failed to deserialize JSON '%s'.", row), e);
+            throw new SeaTunnelJsonFormatException(SeaTunnelJsonFormatErrorCode.JSON_DESERIALIZE_FAILED,
```
Comment (TyrantLucifer) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonSerializationSchema.java]: Remove useless error code.

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/exception/SeaTunnelJsonFormatErrorCode.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.format.json.exception;
+
+import org.apache.seatunnel.common.exception.SeaTunnelErrorCode;
+
+public enum SeaTunnelJsonFormatErrorCode implements SeaTunnelErrorCode {
+    JSON_DESERIALIZE_FAILED("JSON_FORMAT-01", "Failed to deserialize JSON format"),
```
Comment (TyrantLucifer) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/exception/SeaTunnelJsonFormatErrorCode.java]: Remove useless error codes.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonToRowConverters.java [hunk 4]
```diff
@@ -354,7 +359,8 @@ private Object convertField(
         JsonToRowConverter fieldConverter, String fieldName, JsonNode field) {
         if (field == null) {
             if (failOnMissingField) {
-                throw new JsonParseException("Could not find field with name '" + fieldName + "'.");
+                throw new SeaTunnelJsonFormatException(SeaTunnelJsonFormatErrorCode.MISS_FIELD_EXCEPTION,
```
Comment (TyrantLucifer) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonToRowConverters.java]: Remove useless error codes.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**